lib/matplotlib/backend_bases.py

```diff
@@ -3428,7 +3422,13 @@ def add_tool(self, name, tool, group=None, position=None):
                           'not added')
             return
 
-        self._tools[name] = tool_cls(self, name)
```

Comment:
Yes, this ensures that.  I allowed someone to three options to add a tool to the toolbar.
1. Give it the name (string) of the tool that navigation already knows about.
2. Give it a list of [name, class]
3. A tool object (and thus already has a name).

In this third case, we want to ensure that the tool belongs to THIS navigation, in case it was not set before (or got transferred here from another navigation)